docs/_docs/02_features/debugging.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: collection-browser-doc
+title: Debugging
+category: features
+categories_url: features
+excerpt: Learn how to debug issues with terragrunt and terraform.
+tags: ["DRY", "Use cases", "CLI"]
+order: 220
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+## Debugging
+
+Terragrunt and Terraform usually play well together in helping you
+write DRY, re-usable infrastructure. But how do we figure out what
+went wrong in the rare case that they _don't_ play well?
+
+Terragrunt provides a debug mode you can access through the `--terragrunt-debug`
+command flag. For example you could use it like this to debug an `apply`
+that's producing unexpected output:
+
+    $ terragrunt --terragrunt-debug apply
+
+Running this command will do two things for you:
+  - Output a file named `terragrunt-debug.tfvars` to your terraform working
+    directory (the same one containing your `terragrunt.hcl` or `main.tf`
+    terraform code)
```

Comment:
The terragrunt working directory is different from the terraform working directory. Based on our discussions in slack, I believe the intention here was the terragrunt working dir:

```suggestion
  - Output a file named `terragrunt-debug.tfvars` to your terragrunt working
    directory (the same one containing your `terragrunt.hcl`)
```

Reply:
That was probably my wording mistake due to the slack convo, sorry for the confusion.

Reply:
Glad you left this note because I think it's a real misunderstanding! Having vigorous conversation on slack to clarify this and will paste resolution here

Reply:
Result of the vigorous conversation is essentially that Yori's edit is correct. The convo helped me come to a further understanding, first of all, that later versions of `terragrunt` no longer work on infrastructure that lacks a `terragrunt.hcl`! This was a surprise to me since instrumental still has no `terragrunt.hcl` files.

We also learned that, in the case that there is a `terragrunt.hcl` file that does not specify source but instead executes the infra-live module sitting adjacent in the same directory, no `.terragrunt-cache` folder gets created and the terraform working directory becomes the same as the terragrunt one (which was the source of this confusion).